Source/Localhost/LocalhostServer.swift

```diff
@@ -12,68 +12,97 @@ import Criollo
+            guard let routeBlockPopped = self?.popOverlayingRoute(method: method, path: path) else {
+                return
+            }
+            self?.handleRoute(httpMethod: method,
+                              routeBlock: routeBlockPopped,
                               crRequest: req,
                               crResponse: res)
         })
     }
     
     public func delete(_ path: String, routeBlock: @escaping ((URLRequest) -> LocalhostServerResponse?)) {
+        let method: String = "DELETE"
+        self.setOverlayingRoute(method: method, path: path, routeBlock: routeBlock)
         self.server.delete(path, block: {  [weak self] (req, res, next) in
-            self?.handleRoute(httpMethod: "DELETE",
-                              routeBlock: routeBlock,
+            guard let routeBlockPopped = self?.popOverlayingRoute(method: method, path: path) else {
+                return
+            }
+            self?.handleRoute(httpMethod: method,
+                              routeBlock: routeBlockPopped,
                               crRequest: req,
                               crResponse: res)
         })
     }
-
+    
     public func put(_ path: String, routeBlock: @escaping ((URLRequest) -> LocalhostServerResponse?)) {
+        let method: String = "PUT"
+        self.setOverlayingRoute(method: method, path: path, routeBlock: routeBlock)
         self.server.put(path, block: {  [weak self] (req, res, next) in
-            self?.handleRoute(httpMethod: "PUT",
-                              routeBlock: routeBlock,
+            guard let routeBlockPopped = self?.popOverlayingRoute(method: method, path: path) else {
+                return
+            }
+            self?.handleRoute(httpMethod: method,
+                              routeBlock: routeBlockPopped,
                               crRequest: req,
                               crResponse: res)
         })
     }
-    
+
     public func patch(_ path: String, routeBlock: @escaping ((URLRequest) -> LocalhostServerResponse?)) {
+        let method: String = "PATCH"
+        self.setOverlayingRoute(method: method, path: path, routeBlock: routeBlock)
         self.server.add(path, block: {  [weak self] (req, res, next) in
-            self?.handleRoute(httpMethod: "PATCH",
+            self?.handleRoute(httpMethod: method,
                               routeBlock: routeBlock,
                               crRequest: req,
                               crResponse: res)
-            }, recursive: false, method: .patch)
+        })
     }
-    
+
     public func head(_ path: String, routeBlock: @escaping ((URLRequest) -> LocalhostServerResponse?)) {
+        let method: String = "HEAD"
+        self.setOverlayingRoute(method: method, path: path, routeBlock: routeBlock)
         self.server.head(path, block: {  [weak self] (req, res, next) in
-            self?.handleRoute(httpMethod: "HEAD",
+            self?.handleRoute(httpMethod: method,
```

Comment:
Is there a reason for not using the same guard statement as in GET, DELETE, POST and PUT?

Reply:
Will fix it